### PR TITLE
Split scan workflow into seperate CodeQL, FOSSA, Snyk jobs

### PR DESCRIPTION
**Pull request by hiddeco**
So that it also runs on the `main` branch and can properly compare.

Fixes:

> Warning: Code scanning cannot determine the alerts introduced or fixed by this pull request, because 1 analysis was not found.

https://github.com/fluxcd/flux2/runs/1926439471

### CODE REVIEW

**Comment by hiddeco** [comment]
@stefanprodan this does not seem to work as

```
Error: Aborting upload: only one run of the codeql/analyze or codeql/upload-sarif actions is allowed per job
Error: Aborting upload: only one run of the codeql/analyze or codeql/upload-sarif actions is allowed per job
```

I think we need to split them again.

**Comment by stefanprodan** [comment]
> only one run of the codeql/analyze or codeql/upload-sarif actions is allowed per job

😢 but way... each task has an unique id 

Ok let's pull Snyk out.

**Comment by hiddeco** [comment]
I think it would be better to have them all split in that case, as there is little FOSSA and CodeQL have in common (while Snyk and CodeQL do).

File: .github/workflows/bootstrap.yaml [hunk 1]
```diff
@@ -2,8 +2,9 @@ name: bootstrap
 
 on:
   push:
-    branches:
-      - '*'
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```
Comment (stefanprodan) [inline .github/workflows/bootstrap.yaml:7]: Please undo this, we can't share our GH admin token with contributor.

**Review by stefanprodan — CHANGES_REQUESTED** [review]

**Review by stefanprodan — APPROVED** [review]

**Pull request merged**